newlib/libc/misc/lock.c

```diff
@@ -81,7 +81,9 @@ subroutines are required for linking multi-threaded applications.
 
 /* dummy lock routines and static locks for single-threaded apps */
 
-#ifndef __SINGLE_THREAD__
+#include <newlib.h>
```

Comment:
You need to include `newlib.h` otherwise you don't have access to the defined macros like `__SINGLE_THREAD__`